tool/wiki/sync_runtime_parameter.py

```diff
@@ -0,0 +1,159 @@
+                   "NoMin_long":"None", "NoMax_long":"None",
+                   "NoDef_str":"None", "Useless_str":"None",
+                   "Useless_bool":"None",
+                   "true":"1", "false":"0",
+                   "__DBL_MAX__":"1.79769313e+308", "Eps_double":"2.22507386e-308"
+                 }
+
+
+
+#====================================================================================================
+# Classes
+#====================================================================================================
+class parameter():
+    def __init__( self, string ):
+        name, description = self.get_name_description( string )
+        self.name         = name
+        self.link_name    = name
+        self.default      = []
+        self.minimum      = []
+        self.maximum      = []
+        self.description  = description
+        self.NAdd         = 0
+
+    def get_name_description( self, string ):
+        words       = string.split()
+        name        = words[0]
+        description = ""
+        for i in range(3, len(words)):
+            description += words[i] + ' '
+        return name, description[:-1]
+
+    def append_description( self, string ):
+        words = string.split()
+        for i in range(1, len(words)):
+            self.description += ' ' + words[i]
+        return
+
+    def get_link_name( self, file_dict ):
+        for file_name in file_dict:
+            search_pattern = "[%s](#%s)"%(self.name, self.name)
+            if search_pattern in file_dict[file_name]:
+                self.link_name = "[[ %s \\| %s#%s ]]"%(self.name, file_name, self.name)
+                return True
+        return False
+
+
+
+#====================================================================================================
+# Main
+#====================================================================================================
+with open( ALL_PARAM_FILE, 'r' ) as f:
+    lines = f.readlines()
+
+with open( PARAM_CPP_FILE, 'r' ) as f:
+    lines_cpp = f.readlines()
+
+link_source_mds = {}
+for file_path in LINK_FILES:
+    file_name = file_path.split('/')[-1][:-3] # get the file name without path and trailing `.md`
+    with open( file_path, 'r' ) as f:
+        link_source_mds[file_name] = f.read()
+
+params = {}
+
+# get all parameters from ALL_PARAM_FILE
+# NOTE: assuming the line starts with space should be the comment of the above line
+last_key = ""
+for i, line in enumerate(lines):
+    if line[0] == '\n': continue
+    if line[0] == '#': continue
+    if line[0] == ' ':
+        params[last_key].append_description( line )
+        continue
+    last_key = line.split()[0]
+    params[last_key] = parameter( line )
+
+# get all default, min, max, value from PARAM_CPP_FILE
+for i, line in enumerate(lines_cpp):
+    if "ReadPara->Add" not in line: continue
+    words = list( filter( None, re.split( ',| ', line ) ) )
+    if "//" == words[0]:  continue
```

Comment:
Maybe this line could be moved to the front for more efficiency?

Reply:
The variable `words` in the `if` statement is generated by the line `112`. I can use `line.strip()` at the start of the for loop to remove spaces at the beginning, but I still need to separate the string later, so I would like to manipulate the string once only for cleaner code.

Reply:
I first think it can be like line `102` since the file `Init_Load_Parameter.cpp` has no space at the line beginning if the line is a pure comment line.
For example, ```if line[0] == "//": continue``` at line 111.
But I agree with you that some lines maybe have space at the beginning. Your method is safer.